src/os/ogc/query/ogcquery.js

```diff
@@ -0,0 +1,126 @@
+goog.module('os.ogc.query');
+
+const query = goog.require('os.query');
+const ConfirmUI = goog.require('os.ui.window.ConfirmUI');
+const registry = goog.require('os.ogc.registry');
+const {directive: ogclistdirective} = goog.require('os.ui.ogc.List');
+
+
+/**
+ * Generic UI and handlers for an OGC Query
+ */
+class Query {
+  /**
+   * TYPE is required for service to work; override this and hard-code "super(type);" in the constructor
+   * @param {!string} type
+   */
+  constructor(type) {
+    /**
+     * @type {string}
+     */
+    this.type = type;
+
+    /**
+     * @type {string}
+     * @protected
+     */
+    this.eventType_ = 'ogcQuery:*';
+  }
+
+  /**
+   * Handle a area being chosen in the import area border dialog.
+   * @param {function(!ol.Feature, !ol.Feature)} callback The callback to fire when the area is loaded
+   * @param {Array<!ol.Feature>} datum The requested area
+   * @private
+   */
+  onChoice_(callback, datum) {
+    if (datum && datum.length > 0) {
+      const service = registry.getInstance().get(this.type);
+      if (service) {
+        // inject the requested area into the callback so it has some additional context
+        service
+            .get(datum[0])
+            .then(goog.partial(callback, datum));
+      }
+    }
+  }
+
+  /**
+   * Handle area being loaded from the server. Insert into the Area manager.
+   * @param {!ol.Feature} datum The requested area
+   * @param {!ol.Feature} feature The area border feature
+   * @private
+   */
+  onLoaded_(datum, feature) {
+    const service = registry.getInstance().get(this.type);
+    if (service) {
+      feature = service.populateFeature(datum, feature);
+      feature.getGeometry().osTransform();
+      query.addArea(feature);
+    }
+  }
+
+  /**
+   * Launch a confirmation dialog to pick area to import.
+   * @param {!string} title
+   * @param {!string} message
+   * @param {string=} opt_icon
+   * @param {function(!ol.Feature, !ol.Feature)=} opt_callback Callback to fire when the area is chosen and available
+   */
+  launchImport(title, message, opt_icon, opt_callback) {
+    if (!ogclistdirective) return; // touch the <ogclist> directive to get it to inject...
```

Comment:
This line shouldn't be necessary (also `return` statements at the start of functions are make Greg sad). The directive is registered to Angular by that file being included. In order to get around the linter/compiler error for an unused variable, simply don't assign the return value from the `goog.require`

Reply:
Updated to have only the require.